benchexec/tools/hornix.py

```diff
@@ -0,0 +1,40 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+# SPDX-FileCopyrightText: 2024 Jan Kofron <jan.kofron@d3s.mff.cuni.cz>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import benchexec
+import benchexec.result as result
+
+
+class Tool(benchexec.tools.template.BaseTool2):
+    """
+    Info object for the tool Hornix.
+    """
+
+    REQUIRED_PATHS = ["."]
+
+    def executable(self, tool_locator):
+        return tool_locator.find_executable("hornix")
+
+    def version(self, executable):
+        return self._version_from_tool(executable, arg="--version")
+
+    def name(self):
+        """The human-readable name of the tool."""
+        return "Hornix"
+
+    def project_url(self):
+        return "https://github.com/d3sformal/hornix"
+
+    def determine_result(self, run):
+        for line in run.output:
+            if "sat" == line.strip():
+                return result.RESULT_TRUE_PROP
+            elif "unsat" == line.strip():
+                return result.RESULT_FALSE_REACH
+            else:
+                return result.RESULT_UNKNOWN
```

Comment:
Note that this will return unknown at the first line of the output. Is this intended? Then you do not need a loop at all.

Additionally, do you have a way to distinguish between successful tool execution but without a verdict found and an error? Then I recommend to distinguish these in the results as well. This is much more helpful for analyzing the results.

Reply:
Thanks for the comment. True, we do not need the loop here (we have adapted the korn.py script and this has slipped through). Nonetheless, we assumed that the output that is obtained by `determine_result()` is the output of our tool, which we restrict to a single line (sat, unsat, or unknown), so this should be correct. Sorry for the troubles, we are trying to submit for the first time. And thanks a lot once more!

Reply:
We have tried to update the result translation to be a bit more informative.